src/components/algolia-search/index.tsx

```diff
@@ -0,0 +1,195 @@
+    }
+  }, [getEnvironmentProps, formRef, inputRef, panelRef])
+
+  /**
+   * Allow focusing of the search input on pressing '/'
+   */
+  useFocusOnKeyClick(inputRef, '/')
+
+  return (
+    <div className={s.root} {...autocomplete.getRootProps({})}>
+      <form
+        ref={formRef}
+        className={s.form}
+        {...autocomplete.getFormProps({ inputElement: inputRef.current })}
+      >
+        <div className={s.inputWrapperPrefix}>
+          <label {...autocomplete.getLabelProps({})}>
+            <button className={s.submitButton} type="submit" title="Submit">
+              <IconSearch16 />
+            </button>
+          </label>
+        </div>
+        <input
+          className={s.input}
+          ref={inputRef}
+          {...autocomplete.getInputProps({ inputElement: inputRef.current })}
+        />
+        <div className={s.inputWrapperSuffix}>
+          <button
+            className={s.clearButton}
+            title="Clear"
+            type="reset"
+            hidden={!autocompleteState.query}
+          >
+            <IconX16 />
+          </button>
+        </div>
+        {autocompleteState.query ? null : (
+          <IconSlashSquare16
+            className={s.keyboardHintIcon}
+            title="Type '/' to Search"
+          />
+        )}
+      </form>
+
+      {autocompleteState.isOpen && (
+        <div
+          ref={panelRef}
+          className={s.panel}
+          {...autocomplete.getPanelProps({})}
+        >
+          <div className={s.panelLayout}>
+            {autocompleteState.collections.map((collection, index) => {
+              const { source, items } = collection
+
+              return (
+                <section key={`source-${index}`}>
+                  {items.length > 0 && (
+                    <ul className={s.list} {...autocomplete.getListProps()}>
+                      {items.map((item) => {
+                        return (
+                          <li
+                            key={item.objectID}
+                            className={s.item}
+                            {...autocomplete.getItemProps({ item, source })}
+                          >
+                            <HitWrapper
+                              hit={item}
+                              className={s.itemLink}
+                              onHitClick={() => {
+                                autocomplete.setQuery('')
+                              }}
+                              getHitLinkProps={getHitLinkProps}
+                            >
+                              <ResultComponent hit={item} />
+                              <IconCornerDownLeft16 className={s.enterIcon} />
+                            </HitWrapper>
+                          </li>
+                        )
+                      })}
+                    </ul>
+                  )}
+                </section>
+              )
+            })}
+          </div>
```

Comment:
It would probably be reasonable to abstract this part out. I might end up doing that as I work through the mobile UI. 💭